README.md

```diff
@@ -45,8 +45,110 @@ docker run -v "$PWD/myschema.json:/app/schema.json" -p "3000:3000" jormaechea/op
 - [x] Response selection using request header: `Prefer: statusCode=XXX` or `Prefer: example=name` 
 - [x] Request and response logging
 - [x] Servers basepath support
+- [x] Support x-faker and x-count extension methods to customise generated responses
 - [ ] API Authentication
 
+## Customizing Generated Responses
+The OpenAPI specification allows custom properties to be added to an API definition in the form of _x-*_.
+OpenAPI Mocker supports the use of two custom extensions to allow data to be randomised which should allow for more 
+realistic looking data when developing a UI against a mock API for instance.
+
+### x-faker
+The _x-faker_ extension is valid for use on properties that have a primitive type (e.g. `string`/`integer`, etc.) 
+and can be used within an API definition to use one or more methods from the excellent 
+[faker](https://www.npmjs.com/package/faker) library for generating random data.
+
+Given the following API definition:
+```yaml
+openapi: '3.0.2'
+info:
+  title: Cats
+  version: '1.0'
+servers:
+  - url: https://api.cats.test/v1
+paths:
+  /cat:
+    get:
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  firstName:
+                    type: string
+                    x-faker: name.firstName
+                  lastName:
+                    type: string
+                    x-faker: name.lastName
+                  fullName:
+                    type: string
+                    x-faker: '{{name.firstName}} {{name.lastName}}'
+                  age:
+                    type: string
+                    x-faker: 'random.number({ "min": 1, "max": 20 })'
+                  
+``` 
+
+A JSON response similar to the following would be produced:
+```JSON
+{
+    "firstName": "Ted",
+    "lastName": "Kozey",
+    "fullName": "Herbert Lowe",
+    "age": 12
+}
+```
+
+The _x-faker_ extension accepts values in 3 forms:
+1. _fakerNamespace.method_. e.g. `random.uuid`
+2. _fakerNamespace.method({ "methodArgs": "in", "json": "format" })_. e.g. `random.number({ "max": 100 })`
+3. A mustache template string making use of the 2 forms above. e.g. `My name is {{name.firstName}} {{name.lastName}}`
+
+*NOTE*: To avoid new fake data from being generated on every call, up to 10 responses per endpoint is cached 
+that is based on the incoming query string, request body and headers.
```

Comment:
```suggestion
*NOTE*: To avoid new fake data from being generated on every call, up to 10 responses per endpoint are cached 
based on the incoming query string, request body and headers.
```

Reply:
This is what I get for rushing to commit these changes at half 1 in the morning... apologies.